pkg/volume/glusterfs/glusterfs_util.go

```diff
@@ -0,0 +1,58 @@
+package glusterfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+
+	"github.com/golang/glog"
+)
+
+// readGlusterLog will take the last 2 lines of the log file
+// on failure of gluster SetUp and return those so kubelet can
+// properly expose them
+// return nil on any failure
+func readGlusterLog(path string, podName string) error {
+
+	var line1 string
+	var line2 string
+	linecount := 0
+
+	glog.Infof("glusterfs: failure, now attempting to read the gluster log for pod", podName)
+
+	// Check and make sure path exists
+	if len(path) == 0 {
+		glog.Errorf("glusterfs: log file does not exist for pod: %s", podName)
+		return nil
+	}
+
+	// open the log file
+	file, err := os.Open(path)
+	if err != nil {
+		glog.Errorf("glusterfs: could not open log file for pod: %s", podName)
+		return nil
+	}
+	defer file.Close()
+
+	// read in and scan the file using scanner
+	// from stdlib
+	fscan := bufio.NewScanner(file)
+
+	// rather than guessing on bytes or using Seek
+	// going to scan entire file and take the last two lines
+	// generally the file should be small since it is pod specific
+	for fscan.Scan() {
+		if linecount > 0 {
+			line1 = line2
+		}
+		line2 = fscan.Text()
+		
+		linecount++
+	}
+
+	if linecount > 0 {
+		return fmt.Errorf("%v", "\n" + line1 + "\n" + line2 + "\n")
+	}
+	return nil
+}
```

Comment:
@rootfs  - the log data gets exposed in the kubelet.go around line 1793 - this is where and when the volume events are generated so the user will see the error from the log in the pod describe event.   See the *improved behavior* excerpt from initial comment above for what it looks like - if that is what you were asking?